### ISSUE TITLE
Custom downstream cluster secrets remain in the local cluster after the custom downstream cluster is deleted

### ISSUE THREAD

**Issue by sbstp** [rancher/rancher#40849]
**Rancher Server Setup**
- Rancher version: v2.6.8, v2.6.9, v2.7.1 (any)
- Installation option (Docker install/Helm Chart): docker
- Proxy/Cert Details: nothing specific

**Information about the Cluster**
- Kubernetes version: any
- Cluster Type (Local/Downstream): downstream
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider): custom

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom)
  - If custom, define the set of permissions: admin

**Describe the bug**

Custom downstream cluster secrets remain in the local cluster after the custom downstream cluster is deleted.

**To Reproduce**

* Create a custom downstream RKE1 cluster
* Note the cluster ID
* List secrets in the **local** cluster, `kubectl -n cattle-system get secrets -l cattle.io/creator=norman`
* You should see one with the cluster ID
* Delete the custom cluster you created
* List secrets in the **local** cluster, `kubectl -n cattle-system get secrets -l cattle.io/creator=norman`

**Result**
The secret whose name is the deleted cluster ID is still there.

**Expected Result**

The secret whose name is the deleted cluster ID should be removed

**Additional context**

SURE-5388

**Comment by skanakal** [comment]
Yes, this issue is reproducible with Rancher version v2.7.5-rc4.

The following secret was created after the cluster was created:


```
newrancherserver:~ # kubectl get clusters -A
NAMESPACE       NAME      READY   KUBECONFIG
fleet-default   c-l7mkt   true    c-l7mkt-kubeconfig
fleet-local     local     true    local-kubeconfig
newrancherserver:~ # 
newrancherserver:~ # kubectl get secrets -A -l cattle.io/creator=norman | grep c-c-l7mkt
cattle-system        c-c-l7mkt                           Opaque   1      29m
newrancherserver:~ # 
```

Even after deleting the downstream cluster, the secret still exists. The clusterregistrationtoken and namespace were successfully deleted, leaving only the stale object, which is the secret, on the Rancher server:


```
newrancherserver:~ # kubectl get clusters -A
NAMESPACE     NAME    READY   KUBECONFIG
fleet-local   local   true    local-kubeconfig
newrancherserver:~ # 
newrancherserver:~ # kubectl get secrets -A -l cattle.io/creator=norman | grep c-c-l7mkt
cattle-system   c-c-l7mkt   Opaque   1      36m
newrancherserver:~ #
```

2023/06/13 04:12:06 [INFO] Stopping cluster agent for c-l7mkt
2023/06/13 04:12:06 [INFO] Deleting cluster [c-l7mkt]
2023/06/13 04:12:06 [INFO] [mgmt-cluster-rbac-remove] Deleting namespace c-l7mkt
2023/06/13 04:12:06 [ERROR] [etcd-backup] unable to delete backup backup [c-l7mkt-rl-6pqfl]: clusters.management.cattle.io "c-l7mkt" not found
2023/06/13 04:12:06 [ERROR] error syncing 'c-l7mkt/default-token': handler cluster-registration-token: clusters.management.cattle.io "c-l7mkt" not found, requeuing
2023/06/13 04:12:09 [ERROR] error syncing 'c-l7mkt/system': handler cluster-registration-token: clusters.management.cattle.io "c-l7mkt" not found, requeuing
2023/06/13 04:12:09 [ERROR] error syncing 'c-l7mkt/default-token': handler cluster-registration-token: clusters.management.cattle.io "c-l7mkt" not found, requeuing
2023/06/13 04:12:15 [ERROR] error syncing 'c-l7mkt/m-b3a2d6c85940': handler node-controller-sync: Operation cannot be fulfilled on nodes.management.cattle.io "m-b3a2d6c85940": StorageError: invalid object, Code: 4, Key: /registry/management.cattle.io/nodes/c-l7mkt/m-b3a2d6c85940, ResourceVersion: 0, AdditionalErrorMsg: Precondition failed: UID in precondition: a5fcdf53-3f26-47bc-a7fb-c6b01a00aa8d, UID in object meta: , requeuing

**Comment by pratikjagrut** [comment]
Hi, @Jono-SUSE-Rancher We have got the fix ready for this issue. Can you PTAL and add it to the milestone?

**Comment by gaktive** [comment]
Moving to Hostbusters board since all feedback indicates that this should belong here.

Putting to 2.10.0 but I'd endorse this as a backport candidate to 2.9.x since this has been kicking around for a while.

**Comment by Josh-Diamond** [comment]
Ticket #40849 - Test Results - ✅ 
---

Verified on Rancher `v2.11-e9f0646414809cbaee8a1de474d8da152b522f06-head`:

| Scenario | Test Case | Result |
|---|---|---|
| 1. | Provision a downstream RKE1 Custom cluster => delete cluster => ensure associated secret is local cluster is removed | ✅ |

---

Scenario 1 - ✅ 
1. Fresh install of `v2.11-head`
2. Provision a downstream RKE1 Custom cluster
3. Verify associated secret is created in the local cluster
4. Once verified, delete the cluster
5. Verify associated secret in the local cluster is removed
6. Verified - local cluster secret associated with downstream cluster is successfully removed; as expected

Screenshot:
<img width="582" alt="Screenshot 2025-01-09 at 9 43 11 AM" src="https://github.com/user-attachments/assets/32089fd9-c548-43fa-acea-e60903667992" />
